hook!-full-game/src/Game.gd

```diff
@@ -1,5 +1,22 @@
 extends Node
 
+onready var player : Player = $Player
+
+
+func _ready() -> void:
+	Events.connect("player_died", self, "_on_Events_player_died")
+	CheckpointUtils.mark_visited_checkpoints()
+	var last_visited_position := Vector2(CheckpointUtils.checkpoints.last_visited_x,\
+		CheckpointUtils.checkpoints.last_visited_y)
+	if last_visited_position != Vector2.ZERO:
+		player.global_position = last_visited_position
+
+
 func _unhandled_input(event: InputEvent) -> void:
 	if event.is_action_pressed("restart"):
 		get_tree().reload_current_scene()
+
+
+func _on_Events_player_died() -> void:
+	yield(get_tree().create_timer(Defaults.DEATH_ANIMATION_TIME), "timeout")
```

Comment:
You shouldn't split the logic for the death transition and respawn into multiple nodes. It makes the code hard to keep track of. The code that describes the transition sequence should all be in the same place and be linear.